### PR TITLE
Nerfs the RCD

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

Walls now require you to also destroy their girder to deconstruct them with an RCD.
Airlocks now require you to also destroy their assembly if you want to deconstruct them.
You can no longer build walls directly, instead you have to build a girder and then a wall.
Slightly adjusted time and resources required for these tasks.

Also two fixes:
Split second board insertion no longer breaks RCD air alarm construction.
Split second door reinforcement now prevents RCDs from deconstructing an airlock.

## Why It's Good For The Game

This is in pursuit of making walls and rooms in general mean more. Currently, they are way too weak and there are a lot of gamey tools to just quickly break them.

This is also the source of the r-wall spam we are seeing everywhere when people are mapping. If normal walls weren't quite as pathetic as they currently are, we surely wouldn't see as much of that.

## Changelog
:cl:
balance: RCDs now have an extra step when deconstructing airlocks and walls.
balance: RCDs now have to construct a girder before they can construct a wall.
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
time to invest in thermite

**Comment by Akrilla** [comment]
slightly faster construction device

**Comment by dr-bee** [comment]
What is the point of using this over a welder if you do this? how is this a special tool at all?

I understand the desire for balance but making this change and keeping the inefficient cost of the RCD is lame.

**Comment by nemvar** [comment]
The point is that it is convinient. Not that it is faster.

How is this a special tool at all? Do you know how long normal airlock deconstruction takes? This is still miles faster. The RCD is probably the single strongest items in the game. You still skip on one deconstruction step for walls.

What is it with you people and not wanting walls to be good? I don't know a single person that hasn't complained about r-wall spam on mapping and if someone tries to solve the problem at the root everyone loses their mind due to all the bikeshedding.

TL;DR Your criticism is stupid.

**Comment by MMMiracles** [comment]
but the rcd was originally buffed so it would be faster to build/deconstruct, not be more convenient. rooms don't mean anything because there's like 10 of the fuckers you can get easily and more are printable round start if you so desire.

**Comment by nemvar** [comment]
Have you ever seen 10 people actually coordinate? You are being silly right now. Increasing the time that it takes to deconstruct a wall does directly correlate to how much it means to break into said room as time is the deciding factor for that.

If resource usage and time investment go up, how does that not strengthen the integrity of a thing?

But yeah, the RCD should probably be more restricted than it currently is.

**Comment by Tlaltecuhtli** [comment]
making rebuilding holes more of a pain is dumb    if you think its 2 ez to break in places with it nerf deconning  not building walls??? if you think its 2 ez 2 get remove it from lathe and have them locked behind engi vendor + restocks

**Comment by nemvar** [comment]
Plasmacutters are still slower, so please stfu if you have nothing productive to say. Please stop saying "what about this mostly unrelated thing that is out of the scope of this PR" and focus on the topic at hand. If you have any specific problems with any of these nerfs, please say so.

Hole repair does get slightly harder with the changes to wall construction, but that never the limiting factor in repairing a hole anyways. From my experience, the limiting factors are the missing reward structure for doing so, the dull gameplay and the thousand different kinds of machines/air alarms/piping/APCs and so on.
 I have repaired a bunch of holes without even using an RCD and normal sheet construction works perfectly fine for wall building. Since this simply imitates that process, the strain on the engineers won't increase that much.

**Comment by Ratvar** [comment]
Maybe making walls and floors... should eat less resources? So it's less of a hussle to repair rooms in other way? It'll still take load of time to wall off a place, and you already can do it using metal sheets.

**Comment by nemvar** [comment]
Wall construction only went up from 18 to 20.

**Comment by dr-bee** [comment]
Or make deconstruction get resources instead of using them. that way the tool actually becomes a useful building interface instead of a wasteful piece of shit. Still should be faster than normal tools and probably should be removed from the autolathe and moved to a cargo crate. Actually giving cargo a reason to order something.

**Comment by MCterra10** [comment]
> This is also the source of the r-wall spam we are seeing everywhere when people are mapping.

I feel personally attacked

**Comment by MMMiracles** [comment]
I'm not exactly sure what the r-wall spam comment is supposed to mean since going as far back as when the prison satellite was a thing, secure areas like atmos/bridge/the brig had a fair bit of R-walling. The only real difference is the game has had more high-security areas added to the station rather than a reaction to the addition of more RCDs easily accessible mid-round.

**Comment by MCterra10** [comment]
@MMMiracles I'm fairly sure they're referencing #45641 where they foreshadowed this set of PRs

**Comment by zxaber** [comment]
Hey, while you're in the RCD code, could you make the airlock access window have the unrestricted access settings too? Currently you can only set it if you manually build an airlock and use the airlock electronics.

Also, any chance you could leave the mech RCD as able to instant-build walls? It would give a neat reason to want to use an APLU to fix bombings or build new rooms.

**Comment by diegoxt45** [comment]
this is... terrible what is /tg/ doing?

**Comment by GuyonBroadway** [comment]
I feel that your crusade to make walls "more meaningful" is unecessary. 

I like the state it is right now with regular walls bypassable with a little effort and rwalls being the tougher stuff.

I disagree with your opinion on the subject.

**Comment by halitosisman** [comment]
What problem does the wall girder requirement solve? The long time it takes to rebuild departments or the time requirement for any large construction project?

**Comment by CRITAWAKETS** [comment]
Oh hey, it even CONSTRUCTS girders! Why not just carry around metal and tools and just ditch the RCD? You should just rename it to Rapid Airlock Device if thats what you want the RCD to become.

Also did you ignore engiborgs being able to use RCDs to decon r-walls?

**Comment by SpinnerMaster** [comment]
Ah yes, meaningful walls, just the content I was looking for.

Instead of removing functionality, just make it use more matter/take longer when there's no girders in place already or if the wall/airlock has already been partially deconstructed.

**Comment by Cyprex** [comment]
Engineering needs to be able to create and remove walls with relative ease, right now you're just making building, which is already slow, even slower. If you want to make walls stronger against greytide you need to give engineers special tools to make building and removal as easy as before.

**Comment by Xist3nce** [comment]
From a design perspective, you need to look at the playerbase to see what would be best, and not go with odd emotions. Walls are not strong, this is a fact. However engineers should have the ability to rapidly put rooms back together. Takes 1 syndicate agent 30 seconds to rig up a bomb that takes half of medbay out. If it takes 10 minutes to put medbay back together, there's a huge power differential. Not that antagonists should be in line destruction/construction wise with engineers, but when you look at that huge disparity, with the average time the server's rounds last, it just does not seem feasible. Engineering already spends the first 10 of a round setting up SM, Solars, whatever other engine toys. MRP servers should have rules against the tide, so I won't address them. Now the LRP servers last usually 30 - 45 minutes, sometimes more when later. So 10 minutes prior set up, one bomb? 10 more minutes. 2? 10 more. Shuttle called? Try and make the escape hall less swiss cheese until people can get out and that's the whole shift. The average traitor only usually gets one of those if at all, so not so bad, Nuke ops? Chances are 50% of the station will be destroyed. Engineers should be able to at least make the major halls of the station livable in a viable amount of time. 

Game design isn't about feelings, I understand your beef with the walls. I really do, but it has absolutely no basis in the reality of the game and how majority of the playerbase sees it. Walls could do with a buff, but making them shouldn't be a hassle. The RCD was introduced because making the walls manually was slow, unwieldy, and terrible for a real server with a real playerbase. The walls themselves have just under a good durability in that only people with the proper tools and skills can get in. Maybe leave RCDs and engineering alone, as they have the tools for this, and look towards the more widely available tools. Maybe increase the time it takes the toolbelt tools to get through walls, maybe make the walls have more durability, but taking away the ability to damage control seems counter-intuitive.

**Comment by Ratvar** [comment]
Or maybe just nerf wall/door removing ability, because that's the problem and is unrelated to repairing areas, and leave everything else as it was.

**Comment by GuyonBroadway** [comment]
I don't agree that door and wall deleting is a problem. AI sat sieges are currently balanced around RCD power and any captain that lets a CE RCD his way into his office and loot the place before he can take everything important frankly isn't robust enough to have any meaningful input on balancing.

**Comment by CRITAWAKETS** [comment]
Guyon, outside of the RCDs that engiborgs have, RCDs cant decon rwalls.

The AI Sat and Captain's Office have only r-walls. If you are talking about deconning the airlocks, who the fuck does that. For the captain's office you can just hack in and for the AI Sat deconning the airlocks puts you in the way of the turrets.

**Comment by cacogen** [comment]
it's nice not being the one with that thumb ratio and those laughing emojis

**Comment by wesoda25** [comment]
Windows and walls are both a fucking joke. Its easy as fuck to get pass them, and hardly provide an obstacle. Windows are now way fucking stronger than walls, and this is just one PR set to make walls strong again. 

That being said, I think the window buff was much too harsh, and should be toned down a little, with walls made a decent bit harder to get past.

**Comment by Yttriums** [comment]
Just take RCD/RPD out of hacked autolathes and put them in the engi protolathe or cargo. It doesn't make sense that this advanced builder can be constructed with only iron anyways

**Comment by Akrilla** [comment]
RCDs are not trivial to get a hold of, only engineers have them, and tends to remain that way. If a traitor engineer uses an RCD then that's the outcome of an engineer becoming a traitor.

**Comment by Shadowflame909** [comment]
I'm pretty sure the maintainer direction agrees that it's too easy to blow the station to smithereens but it takes too long to repair it.

This nerf is not wararnted and we need to be able to deconstruct some bad boy walls and airlocks fast when they've been tampered with

Looking at Malf Ai's. The hijacking the shuttle gimmick of emag + bolts. Or even worse, when that wizard is spamming blink.

**Comment by Vasily2013** [comment]
> RCDs are not trivial to get a hold of, only engineers have them, and tends to remain that way. If a traitor engineer uses an RCD then that's the outcome of an engineer becoming a traitor.
> 
> edit: Oh yeah I forgot hacked autolathe - maybe just remove it out of that and be done with it.

This man speaks the truth. RCDs are hard to get in general, that's part of the reason that they can be so powerful and not break the game apart. A traitor is better setting up an X4 charge or thermite normally. Both are in my view as hard to get or easier to access than an RCD.

**Comment by Ratvar** [comment]
This man also spoke about easy to get cargo autolathe RCDs, which you ignored. And thermite's getting nerfed. X4 invest your valuable telecrystals/are a clear sign of tator tot, and, if they are too good at destroying walls, they'll probably get a nerf like all easy ways to punch walls and doors down are getting right now.

**Comment by actioninja** [comment]
Not sure about this one. I like the deconstruction changes and I get that the building changes are just to pair with them, but building should be very easy imo.
How about something like forkknife construction? you build holograms of things which then autoconstruct to a real wall over the course of about a minute, and they can be pretty easily beaten down while this is happening. Makes it still easy to build with but griefing a lot harder.

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
something above is fine, but I'd like construction to not be nerfed big/small and the focus be on deconstruction (which is something i'm ok with merging conceptually).

**Comment by nemvar** [comment]
The big problem I was seeing with not nerfing construction in tandem is that you can also grief with construction. Imagine if deconstruction was extremely hard but you could create a wall in 3 seconds.

Anyways, I had a better idea for a nerf but the real question if I can be bothered implementing it. Maybe later.
